### PR TITLE
[Estimation] Dynamic reset functionality for ExtendedKalmanFilter

### PR DESCRIPTION
**Pull request by prashanthr05**
This PR adds a method to dynamically reset the EKF components through a function call.
Also, includes a few minor changes
 - added a ignore function template to ignore unused parameters
 - added an ekfComputeJacobian method to also take into account inputs during computation
 - removed stray variable

### CODE REVIEW

**Comment by prashanthr05** [comment]
@traversaro if you could review this PR, it would be great. Thank you!

File: src/estimation/include/iDynTree/Estimation/AttitudeQuaternionEKF.h [hunk 1]
```diff
@@ -253,6 +253,7 @@ namespace iDynTree
          * @return bool true/false if successful or not
          */
         bool ekfComputeJacobianF(iDynTree::VectorDynSize& x, iDynTree::MatrixDynSize& F) override;
+        bool ekfComputeJacobianF(iDynTree::VectorDynSize& x, iDynTree::VectorDynSize& u, iDynTree::MatrixDynSize& F) override;
```
Comment (traversaro) [inline src/estimation/include/iDynTree/Estimation/AttitudeQuaternionEKF.h]: What is `u`? Could it be documented in the doxygen docs? It probably make sense to just document the variant of the method with more arguments, and for the doxygen of the method with less arguments just refer to the method with more arguments using an appropriate doxygen command like http://www.doxygen.nl/manual/commands.html#cmdsa .

File: src/estimation/include/iDynTree/Estimation/ExtendedKalmanFilter.h [hunk 2]
```diff
@@ -161,6 +163,21 @@ namespace iDynTree
...
+         *             where the size of the system state or the measurements keep evolving with time
+         *
+         */
+        bool ekfReset(const size_t& state_size,
+                                 const size_t& input_size,
+                                 const size_t& output_size,
+                                 const iDynTree::Span<double>& x0,
+                                 const iDynTree::Span<double>& P0,
+                                 const iDynTree::Span<double>& Q,
+                                 const iDynTree::Span<double>& R);
```
Comment (traversaro) [inline src/estimation/include/iDynTree/Estimation/ExtendedKalmanFilter.h]: Q and R are typically matrices, what is the expected size of the Span<double> to represent them? Is the ordering column major or row major?

File: src/estimation/include/iDynTree/Estimation/ExtendedKalmanFilter.h [hunk 3]
```diff
@@ -254,7 +271,14 @@ namespace iDynTree
...
 
-    private:
+        /**
+        * function template to ignore unused parameters
+        */
+        template <typename T>
+        void ignore(T &&)
+        { }
+
+            private:
```
Comment (traversaro) [inline src/estimation/include/iDynTree/Estimation/ExtendedKalmanFilter.h]: Shouldn't this be inline with the `public` keywork that is already in the class?

File: src/estimation/include/iDynTree/Estimation/ExtendedKalmanFilter.h [hunk 4]
```diff
@@ -254,7 +271,14 @@ namespace iDynTree
          */
         bool ekfGetStateCovariance(const iDynTree::Span<double> &P) const;
 
-    private:
+        /**
+        * function template to ignore unused parameters
+        */
+        template <typename T>
+        void ignore(T &&)
+        { }
```
Comment (traversaro) [inline src/estimation/include/iDynTree/Estimation/ExtendedKalmanFilter.h]: Can't this be private?

File: src/estimation/include/iDynTree/Estimation/ExtendedKalmanFilter.h (hunk 3 above)
Comment (prashanthr05) [reply src/estimation/include/iDynTree/Estimation/ExtendedKalmanFilter.h]: inline you mean in the same intendation?

Comment (traversaro) [reply src/estimation/include/iDynTree/Estimation/ExtendedKalmanFilter.h]: Yes sorry, I actually mean `in line`, not `inline` . : )

**Comment by prashanthr05** [comment]
@traversaro I've addressed your review in the squashed commit.

File: src/estimation/include/iDynTree/Estimation/ExtendedKalmanFilter.h (hunk 4 above)
Comment (prashanthr05) [reply src/estimation/include/iDynTree/Estimation/ExtendedKalmanFilter.h]: If I put this private, it cannot be used by the inheriting classes.

Comment (prashanthr05) [reply src/estimation/include/iDynTree/Estimation/ExtendedKalmanFilter.h]: I will mark it protected

**Comment by prashanthr05** [comment]
@traversaro  Travis seems to have some timeout issues. Could you check it and let me know if I should do something about it?

**Comment by prashanthr05** [comment]
@traversaro friendly ping! :)

**Comment by traversaro** [comment]
I restarted the Travis build, in the meanwhile can you resolve the conflicts on the release notes?  Btw, I think you should have  the powers to restart Travis jobs, let me know if you do not have them.

**Comment by prashanthr05** [comment]
> I restarted the Travis build, in the meanwhile can you resolve the conflicts on the release notes?

Done in https://github.com/robotology/idyntree/pull/553/commits/4e3680d7c5a1b236cbb43acef5de159b3fc1cb15. Thanks!

**Comment by prashanthr05** [comment]
> Btw, I think you should have the powers to restart Travis jobs, let me know if you do not have them.

I think I have them. I see a Cancel Build option on a currently running job.

**Pull request merged**